src/util/scroll.js

```diff
@@ -1,7 +1,37 @@
 import scrollToIndex from './scrollToIndex.js';
 import { getToolState } from '../stateManagement/toolState.js';
 
-export default function (element, images, loop = false) {
+function scrollWithoutSkipping (stackData, pendingEvent, element) {
+  if (stackData.pending[0] === pendingEvent) {
+    if (stackData.currentImageIdIndex === pendingEvent.index) {
+      stackData.pending.splice(stackData.pending.indexOf(pendingEvent), 1);
+
+      if (stackData.pending.length > 0) {
+        scrollWithoutSkipping(stackData, stackData.pending[0], element);
+      }
+
+      return;
+    }
+
+    element.addEventListener('cornerstonenewimage', function(event) {
```

Comment:
Maybe I'm missing it. Is this the right event?

Reply:
I believe it is, as this seems to be the event that is triggered when the index *actually* changes.